modules/javafx.graphics/src/main/java/com/sun/glass/ui/mac/MacAccessible.java

```diff
@@ -806,6 +802,15 @@ public void sendNotification(AccessibleAttribute notification) {
 
     @Override
     protected long getNativeAccessible() {
+        if (this.peer == 0L) {
+            AccessibleRole role = (AccessibleRole) getAttribute(ROLE);
+            if (role == null) role = AccessibleRole.NODE;
```

Comment:
It can be when customer creates a custom component that does not provide accessible attributes. With the new API that will be covered by the default implementation that will report some generic role.

Reply:
OK. It looks like a `null` here can only happen for an accessible object that _isn't_ a `Node`, since `Node.getAccessibleRole` already ensures that a null value is converted to `AccessibleRole.NODE`. As long as the default implementation (of `AccessibleRole.NODE`) does the right thing for an object that doesn't report a specific role and isn't a `Node`, what you have seems fine.